### PR TITLE
Facilitate Jupyter testing with nbmake

### PR DESCRIPTION
**Pull request by thewtex**
With nbmake, the connection file is a temp file such as '/tmp/tmpabc123.json'. Use the name of this file for the kernel id in this case.

### CODE REVIEW

**Comment by thewtex** [comment]
CC @bnmajor

**Review by oeway — APPROVED** [review]
Great! Thanks!

File: python/imjoy_rpc/connection/jupyter_connection.py [hunk 1]
```diff
@@ -29,9 +30,15 @@ def __init__(self, rpc_context):
...
-        ).group(1)
+        connection_file = ipykernel.connect.get_connection_file()
+        if 'kernel-' in connection_file:
+            self.kernel_id = re.search(
+                "kernel-(.*).json", ipykernel.connect.get_connection_file()
+            ).group(1)
+        else:
+            # nbmake
+            self.kernel_id = Path(connection_file).stem
+        print('kernel_id', self.kernel_id)
```
Comment (oeway) [inline python/imjoy_rpc/connection/jupyter_connection.py:41]: Do we want to keep the print statement here?

**Comment by oeway** [comment]
Could you please run `black .` to fix the format error too?

File: python/imjoy_rpc/connection/jupyter_connection.py (hunk 1 above)
Comment (thewtex) [reply python/imjoy_rpc/connection/jupyter_connection.py:41]: ah, no! :-)

**Comment by thewtex** [comment]
> Could you please run black . to fix the format error too?

Yes, done :+1: 

Could we please get a release with this? It will help us get started on testing :green_heart:

**Comment by oeway** [comment]
Great!

> Could we please get a release with this? It will help us get started on testing 💚

Yes,  the CI will make the release if we change the version number in 3 places (VERSION file, pakcage.json and pakcage-lock.json).

**Comment by thewtex** [comment]
@oeway awesome, thanks!

File: python/imjoy_rpc/connection/jupyter_connection.py [hunk 2]
```diff
@@ -29,9 +30,14 @@ def __init__(self, rpc_context):
         self.interface = None
         self.rpc_context = rpc_context
         self._codecs = {}
-        self.kernel_id = re.search(
-            "kernel-(.*).json", ipykernel.connect.get_connection_file()
-        ).group(1)
+        connection_file = ipykernel.connect.get_connection_file()
+        if "kernel-" in connection_file:
+            self.kernel_id = re.search(
+                "kernel-(.*).json", ipykernel.connect.get_connection_file()
```
Comment (MartinHjelmare) [inline python/imjoy_rpc/connection/jupyter_connection.py:36]: Can we not use `connection_file` here in the `re.search` call?

**Pull request merged**